src/test/ui/lint/fn_must_use.stderr

```diff
@@ -11,8 +11,26 @@ note: lint level defined here
    |         ^^^^^^^^^^^^^^^
 
 warning: unused return value of `MyStruct::need_to_use_this_method_value` which must be used
-  --> $DIR/fn_must_use.rs:34:5
+  --> $DIR/fn_must_use.rs:45:5
    |
-34 |     m.need_to_use_this_method_value();
+45 |     m.need_to_use_this_method_value();
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
+warning: unused return value of `EvenNature::is_even` which must be used: no side effects
+  --> $DIR/fn_must_use.rs:46:5
+   |
+46 |     m.is_even(); // trait method!
+   |     ^^^^^^^^^^^^
+
+warning: unused return value of `std::cmp::PartialEq::eq` which must be used
+  --> $DIR/fn_must_use.rs:48:5
+   |
+48 |     2.eq(&3);
+   |     ^^^^^^^^^
+
+warning: unused comparison which must be used
+  --> $DIR/fn_must_use.rs:49:7
+   |
+49 |     2 == 3;
```

Comment:
The result of the comparison is unused. The span should be including the rhs and lhs of the operation